### PR TITLE
Add a justified balance getter to forkchoice

### PR DESCRIPTION
**Pull request by potuz**
This PR adds a handler to forkchoice to request justified balances only when the justified checkpoint has changed instead of receiving this information on every call to Head. At this moment this is a setter function that just sets this handler and an implementation in the stategen package.

### CODE REVIEW

File: beacon-chain/state/stategen/getter.go [hunk 1]
```diff
@@ -59,6 +61,33 @@ func (s *State) StateByRoot(ctx context.Context, blockRoot [32]byte) (state.Beac
...
+	if err != nil {
+		return nil, err
+	}
+	if st == nil || st.IsNil() {
+		return nil, errNilState
+	}
+	epoch := time.CurrentEpoch(st)
+
+	balances := make([]uint64, st.NumValidators())
+	var balanceAccumulator = func(idx int, val state.ReadOnlyValidator) error {
```
Comment (terencechain) [inline beacon-chain/state/stategen/getter.go:77]: why call this an accumulator?

Comment (potuz) [reply beacon-chain/state/stategen/getter.go:77]: you tell me, you wrote that code

Comment (terencechain) [reply beacon-chain/state/stategen/getter.go:77]: Oh well, let's fix it!

File: beacon-chain/forkchoice/interfaces.go [hunk 2]
```diff
@@ -77,4 +77,5 @@ type Setter interface {
 	SetGenesisTime(uint64)
 	SetOriginRoot([32]byte)
 	NewSlot(context.Context, types.Slot) error
+	SetBalancesByRoot(func(context.Context, [32]byte) ([]uint64, error))
```
Comment (kasey) [inline beacon-chain/forkchoice/interfaces.go:80]: Might be a little more clear what this is doing if named `SetBalancesByRooter`?

**Review by kasey — DISMISSED** [review]
Seems like we're just creating the ability to set this and doing so in stategen, not using it yet. LGTM!

**Review by terencechain — APPROVED** [review]

**Pull request merged**